classes/03class/docker/README.md

```diff
@@ -0,0 +1,261 @@
+a70625894939        2 days ago          /bin/sh -c apt-get update                       27.9MB
+5ba0d7847404        2 days ago          /bin/sh -c #(nop) WORKDIR /tmp                  0B    
+ccc6e87d482b        4 weeks ago         /bin/sh -c #(nop)  CMD ["/bin/bash"]            0B    
+<missing>           4 weeks ago         /bin/sh -c mkdir -p /run/systemd && echo 'do…   7B    
+<missing>           4 weeks ago         /bin/sh -c set -xe   && echo '#!/bin/sh' > /…   745B  
+<missing>           4 weeks ago         /bin/sh -c [ -z "$(apt-get indextargets)" ]     987kB 
+<missing>           4 weeks ago         /bin/sh -c #(nop) ADD file:08e718ed0796013f5…   63.2MB 
+```
+
+Considering the behaviour exposed above, it's important to keep the steps that will result in identical state, to be executed before steps that will result in different states. For example, if you're building an image with an application under development and you have the following steps on your image build:
+
+1) Copying application files
+2) Installing application pre-requisites(apache+php) with specific versions
+3) Installing updates on the image
+
+In that case, you can consider that the step 2 will always have the same result. If building the image multiple times a day, the step 3 will most likely have the same result as well. On the other hand, the step one, because is being constantly changed during the development process, will have a different result mostly every build.
+In that case, ordering the steps as 2-3-1 in your Dockerfile, will make Docker reuse the step 2 and 3 for mostly every build, so the build process will be faster and you won't  need to keep downloading and installing the application pre-requites and updates on every build you do.
+
+## Main docker commands
+- docker run
+    - Creates and execute a new container    
+- docker build
+    - Build a new image from a docker file
+- docker exec
+    - Execute a command on a running container
+- docker images
+    - List the images stored on the local system managed by the docker daemon
+- docker ps
+    - List the running containers
+- docker container [action]
+    - Execute an action related to containers
+- docker image [action]
+    - Execute an action related to images
+
+Reference for all docker commands and options: [https://docs.docker.com/engine/reference/commandline/docker/](https://docs.docker.com/engine/reference/commandline/docker/)
+
+## AWS ECS
+[ECS](https://aws.amazon.com/ecs/)(Elastic Container Service) is a fully managed containers orchestration service available on AWS. A container orchestration tool is responsible to coordinate and manage all aspects of the lifecycle of  containers execution. As an example, it can be configured to run 2 replicas of a specific image and whenever something goes wrong with one the replicas, the orchstration tool will notice it and spin up a new container, so it keeps 2 replicas running all the time.
```

Comment:
Are you going to do a walkthrough here? 

If not this might be too theoretical, they might lose track.